version: '3.8'

services:
  mongodb:
    image: mongo
    healthcheck:
      test: ["CMD", "mongo", "--eval", "db.adminCommand('ping')"]
      interval: 10s
      timeout: 10s
      retries: 5
    volumes:
      - mongodb_data:/data/db
    ports:
      - "27017:27017"
    networks:
      - mynetwork

  server:
    build: ./SERVER
    #image: ghcr.io/aziz4785/smartstyle-server:v0.1
    volumes:
      - ./SERVER:/server
    ports:
      - "5000:5000"
    depends_on:
      - mongodb
    networks:
      - mynetwork

  # test:
  #   build: ./SERVER
  #   #image: ghcr.io/aziz4785/smartstyle-test:v0.1
  #   volumes:
  #     - ./SERVER:/server
  #   command: ["sh", "-c", "python -m unittest /server/tests/test_embedding_catalog1.py"]
  #   depends_on:
  #     - mongodb
  #   networks:
  #     - mynetwork

  # apache:
  #   image: httpd:2.4
  #   volumes:
  #     - ./my-httpd.conf:/usr/local/apache2/conf/httpd.conf
  #     - ./my-apache.conf:/usr/local/apache2/conf/apache.conf
  #   ports:
  #     - "80:80"
  #   depends_on:
  #     - server
  #   networks:
  #     - mynetwork

  nginx:
    image: nginx:alpine
    volumes:
      - ./nginx.conf:/etc/nginx/conf.d/default.conf
      - ./SERVER/static:/usr/share/nginx/html/static
      - ./SERVER/templates:/usr/share/nginx/html/templates
    ports:
      - "80:80"
    depends_on:
      - server
    networks:
      - mynetwork
volumes:
  mongodb_data:

networks:
  mynetwork: